### DISCUSSION TITLE
[January 2024] zkSync developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#312] [Announcements]
Greetings devs! Welcome to the first issue of the zkSync developer update. In this monthly series we will showcase different updates about developer tools, integrations, tutorials, developer-oriented content and more. 

Here are the updates for the last month:

**🛠️ Developer tools**

- Etherscan launch: the most popular block explorer is [now available on zkSync](https://twitter.com/zksync/status/1744343399417737266). Start exploring [zkSync on Etherscan](https://era.zksync.network/)
- Remix plugin: in collaboration with Nethermind, we’ve released a plugin for the beloved browser-based IDE so you can build and deploy contracts from your browser. [Read the guide to use the plugin](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3)
- Viem zkSync support: `viem/zksync` contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! Stay tuned for more zkSync support for viem. [Check out the viem zkSync docs](https://viem.sh/zksync)
- zkSync Sepolia testnet integrations: more developer tools now support the zkSync Sepolia testnet:
    - [theGraph CLI](https://www.npmjs.com/package/@graphprotocol/graph-cli): create and deploy subgraphs
    - [BlockPI](https://public.blockpi.io/), [Chainstack](https://chainstack.com/build-better-with-zksync-era/) and [QuickNode](https://www.quicknode.com/chains/zksync) offer RPC nodes.
    - Both [Atlas](https://zksync.atlaszk.com/ide) and [Remix](https://remix.ethereum.org/) allow you to build and deploy smart contracts from your browser

**📖 Documentation**

- New [hello world tutorial](https://docs.zksync.io/build/quick-start/hello-world.html): using `zksync-cli` to deploy a contract on zkSync in less than 5 minutes.
- New [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) with everything you need to know about `zksync-ethers` and how to integrate paymasters in your dApps.
- We also released multiple updates in the zkSync developer documentation, including design improvements and a content restructure. [Check it out here](https://docs.zksync.io/) and let us know what you think!

**📺 Content**

- [In memory node video tutorial](https://www.youtube.com/watch?v=hcavBfyh8JE): learn how to use the in-memory-node to test your contracts and debug transactions.
- [Atlas IDE walkthrough tutorial](https://www.youtube.com/watch?v=PjMXPz2HIoY) with Samee
- New shorts uploaded to the [zkSync Youtube channel](https://www.youtube.com/@zkSync-era/shorts?sub_confirmation=1)

**🔗 Stay connected**

- [Follow @zkSyncDevs on X](https://twitter.com/zkSyncDevs)
- Join the `#dev-general` channel in our Discord

Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by dzhamilrustamov** [comment]
Good

**Reply by Winx0** [reply]
lfg

**Reply by Kimchikim12** [reply]
This new update is awesome.

**Reply by microlincoln** [reply]
WONDERFUL DEVELOPER UPDATE

**Reply by shexinyi321** [reply]
nice work

**Comment by HoseinNikmaramOfficial** [comment]
thanks mate 👍

**Reply by JAY24666** [reply]
Cool update. Looking forward to see more features. Great work team❤️👍

**Comment by waiwai2** [comment]
Hello Antonio

a huge thank you for kicking off the zkSync developer update series. 

The launch of Etherscan for zkSync is a monumental step forward, providing much-needed visibility and ease of exploration for developers and users alike. 

The new documentation, including the hello world tutorial and the frontend quickstart, are also good resources. 

Being part of the zkSync developer community is a rewarding experience.

Thank you for your hard work and dedication.

waiwai2

**Comment by mahdi00007** [comment]
> Greetings devs! Welcome to the first issue of the zkSync developer update. In this monthly series we will showcase different updates about developer tools, integrations, tutorials, developer-oriented content and more.
> 
> Here are the updates for the last month:
> 
> **🛠️ Developer tools**
> 
> * Etherscan launch: the most popular block explorer is [now available on zkSync](https://twitter.com/zksync/status/1744343399417737266). Start exploring [zkSync on Etherscan](https://era.zksync.network/)
> * Remix plugin: in collaboration with Nethermind, we’ve released a plugin for the beloved browser-based IDE so you can build and deploy contracts from your browser. [Read the guide to use the plugin](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3)
> * Viem zkSync support: `viem/zksync` contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! Stay tuned for more zkSync support for viem. [Check out the viem zkSync docs](https://viem.sh/zksync)
> * zkSync Sepolia testnet integrations: more developer tools now support the zkSync Sepolia testnet:
>   
>   * [theGraph CLI](https://www.npmjs.com/package/@graphprotocol/graph-cli): create and deploy subgraphs
>   * [BlockPI](https://public.blockpi.io/), [Chainstack](https://chainstack.com/build-better-with-zksync-era/) and [QuickNode](https://www.quicknode.com/chains/zksync) offer RPC nodes.
>   * Both [Atlas](https://zksync.atlaszk.com/ide) and [Remix](https://remix.ethereum.org/) allow you to build and deploy smart contracts from your browser
> 
> **📖 Documentation**
> 
> * New [hello world tutorial](https://docs.zksync.io/build/quick-start/hello-world.html): using `zksync-cli` to deploy a contract on zkSync in less than 5 minutes.
> * New [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) with everything you need to know about `zksync-ethers` and how to integrate paymasters in your dApps.
> * We also released multiple updates in the zkSync developer documentation, including design improvements and a content restructure. [Check it out here](https://docs.zksync.io/) and let us know what you think!
> 
> **📺 Content**
> 
> * [In memory node video tutorial](https://www.youtube.com/watch?v=hcavBfyh8JE): learn how to use the in-memory-node to test your contracts and debug transactions.
> * [Atlas IDE walkthrough tutorial](https://www.youtube.com/watch?v=PjMXPz2HIoY) with Samee
> * New shorts uploaded to the [zkSync Youtube channel](https://www.youtube.com/@zkSync-era/shorts?sub_confirmation=1)
> 
> **🔗 Stay connected**
> 
> * [Follow @zkSyncDevs on X](https://twitter.com/zkSyncDevs)
> * Join the `#dev-general` channel in our Discord
> 
> Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.
> 
> Keep buidling! 👩‍💻🧑🏻‍💻

good luck

**Comment by montax5** [comment]
Dear Antonio

 Thanks for doing this zkSync developer update .

The advancement of  viem/zksync contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! is great way to take Zksync forward. 
Its my pleasure in helping the development of zksync, love to interact with other devs in discord as well . 
Thanks

**Comment by enerelt0418** [comment]
> Greetings devs! Welcome to the first issue of the zkSync developer update. In this monthly series we will showcase different updates about developer tools, integrations, tutorials, developer-oriented content and more.
> 
> Here are the updates for the last month:
> 
> **🛠️ Developer tools**
> 
> * Etherscan launch: the most popular block explorer is [now available on zkSync](https://twitter.com/zksync/status/1744343399417737266). Start exploring [zkSync on Etherscan](https://era.zksync.network/)
> * Remix plugin: in collaboration with Nethermind, we’ve released a plugin for the beloved browser-based IDE so you can build and deploy contracts from your browser. [Read the guide to use the plugin](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3)
> * Viem zkSync support: `viem/zksync` contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! Stay tuned for more zkSync support for viem. [Check out the viem zkSync docs](https://viem.sh/zksync)
> * zkSync Sepolia testnet integrations: more developer tools now support the zkSync Sepolia testnet:
>   
>   * [theGraph CLI](https://www.npmjs.com/package/@graphprotocol/graph-cli): create and deploy subgraphs
>   * [BlockPI](https://public.blockpi.io/), [Chainstack](https://chainstack.com/build-better-with-zksync-era/) and [QuickNode](https://www.quicknode.com/chains/zksync) offer RPC nodes.
>   * Both [Atlas](https://zksync.atlaszk.com/ide) and [Remix](https://remix.ethereum.org/) allow you to build and deploy smart contracts from your browser
> 
> **📖 Documentation**
> 
> * New [hello world tutorial](https://docs.zksync.io/build/quick-start/hello-world.html): using `zksync-cli` to deploy a contract on zkSync in less than 5 minutes.
> * New [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) with everything you need to know about `zksync-ethers` and how to integrate paymasters in your dApps.
> * We also released multiple updates in the zkSync developer documentation, including design improvements and a content restructure. [Check it out here](https://docs.zksync.io/) and let us know what you think!
> 
> **📺 Content**
> 
> * [In memory node video tutorial](https://www.youtube.com/watch?v=hcavBfyh8JE): learn how to use the in-memory-node to test your contracts and debug transactions.
> * [Atlas IDE walkthrough tutorial](https://www.youtube.com/watch?v=PjMXPz2HIoY) with Samee
> * New shorts uploaded to the [zkSync Youtube channel](https://www.youtube.com/@zkSync-era/shorts?sub_confirmation=1)
> 
> **🔗 Stay connected**
> 
> * [Follow @zkSyncDevs on X](https://twitter.com/zkSyncDevs)
> * Join the `#dev-general` channel in our Discord
> 
> Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.
> 
> Keep buidling! 👩‍💻🧑🏻‍💻

Good luck We have happy long journey

**Comment by DogePlayMid** [comment]
> Greetings devs! Welcome to the first issue of the zkSync developer update. In this monthly series we will showcase different updates about developer tools, integrations, tutorials, developer-oriented content and more.
> 
> Here are the updates for the last month:
> 
> **🛠️ Developer tools**
> 
> * Etherscan launch: the most popular block explorer is [now available on zkSync](https://twitter.com/zksync/status/1744343399417737266). Start exploring [zkSync on Etherscan](https://era.zksync.network/)
> * Remix plugin: in collaboration with Nethermind, we’ve released a plugin for the beloved browser-based IDE so you can build and deploy contracts from your browser. [Read the guide to use the plugin](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3)
> * Viem zkSync support: `viem/zksync` contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! Stay tuned for more zkSync support for viem. [Check out the viem zkSync docs](https://viem.sh/zksync)
> * zkSync Sepolia testnet integrations: more developer tools now support the zkSync Sepolia testnet:
>   
>   * [theGraph CLI](https://www.npmjs.com/package/@graphprotocol/graph-cli): create and deploy subgraphs
>   * [BlockPI](https://public.blockpi.io/), [Chainstack](https://chainstack.com/build-better-with-zksync-era/) and [QuickNode](https://www.quicknode.com/chains/zksync) offer RPC nodes.
>   * Both [Atlas](https://zksync.atlaszk.com/ide) and [Remix](https://remix.ethereum.org/) allow you to build and deploy smart contracts from your browser
> 
> **📖 Documentation**
> 
> * New [hello world tutorial](https://docs.zksync.io/build/quick-start/hello-world.html): using `zksync-cli` to deploy a contract on zkSync in less than 5 minutes.
> * New [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) with everything you need to know about `zksync-ethers` and how to integrate paymasters in your dApps.
> * We also released multiple updates in the zkSync developer documentation, including design improvements and a content restructure. [Check it out here](https://docs.zksync.io/) and let us know what you think!
> 
> **📺 Content**
> 
> * [In memory node video tutorial](https://www.youtube.com/watch?v=hcavBfyh8JE): learn how to use the in-memory-node to test your contracts and debug transactions.
> * [Atlas IDE walkthrough tutorial](https://www.youtube.com/watch?v=PjMXPz2HIoY) with Samee
> * New shorts uploaded to the [zkSync Youtube channel](https://www.youtube.com/@zkSync-era/shorts?sub_confirmation=1)
> 
> **🔗 Stay connected**
> 
> * [Follow @zkSyncDevs on X](https://twitter.com/zkSyncDevs)
> * Join the `#dev-general` channel in our Discord
> 
> Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.
> 
> Keep buidling! 👩‍💻🧑🏻‍💻

Very good

**Comment by sunarus** [comment]
Good luck

**Comment by Irbidonles** [comment]
hay.... any link DC ZkSycn i want to join .......
link on guild and official web of zksync has expied......
please help me......

**Comment by Delaw258** [comment]
Regarding the new hello world tutorial using zksync-cli, it would be beneficial to include some additional details on how developers can leverage this tool for more complex contract deployments or interactions. Providing examples of real-world scenarios or use cases where zksync-cli can streamline development processes would enhance the tutorial's practical value and help developers grasp its potential beyond basic deployments.

**Comment by Adebola-sh** [comment]
hello, you wrote paymasters instead of paymaster, scammers can capitalize on this and use it to lure many people into their scams. this is my observation yet

**Comment by Delaw258** [comment]
Thanks for that observation

**Comment by therntibmaph** [comment]
Great initiative! The monthly developer update provides a comprehensive overview of recent developments, tools, tutorials, and community engagement efforts within zkSync ecosystem. It's crucial for keeping developers informed and engaged, fostering collaboration, and driving adoption. Keep up the good work! 👍🏼

**Comment by mattbrathwaite** [comment]
Great to see the zkSync developer ecosystem thriving with new tools, integrations, tutorials, and content updates! Looking forward to exploring the new features and improvements. Keep building! 👩‍💻🧑🏻‍💻

**Comment by yas3i3** [comment]
developer update provides a comprehensive overview of recent developments, tools, tutorials, and community engagement efforts within zkSync ecosystem.

**Comment by sunarus** [comment]
Nice to see. I want to improve my gaming platform & token in zksync ecosystem

**Comment by kennycobbs** [comment]
zkSync Developer Update:

Exciting new tools and integrations released, including:

Etherscan support for zkSync
Remix plugin for browser-based contract building
Viem support for zkSync transactions
Integration of zkSync Sepolia testnet with various developer tools
📖 Enhanced documentation:

New hello world and frontend quickstart tutorials
Updates and improvements to zkSync developer docs
📺 Engaging content:

Video tutorials on using in-memory node and Atlas IDE
Fresh shorts on zkSync YouTube channel
🔗 Stay connected:

Follow @zkSyncDevs
Join #dev-general channel in Discord
Thanks for being part of zkSync developer community! Keep building! 🚀

**Comment by yas3i3** [comment]
Gitgub @zkSyncDevs gives connection eror.

kennycobbs ***@***.***>, 5 Mar 2024 Sal, 13:19 tarihinde şunu
yazdı:

**Comment by jennifercerso** [comment]
The zkSync team has released several updates and resources for developers, including the launch of Etherscan on zkSync, a new Remix plugin for building and deploying contracts, and viem/zksync support for sending transactions and interacting with contracts via paymasters. The team has also released new tutorials, documentation updates, and content, including a hello world tutorial and in-memory node video tutorial. Developers can stay connected with the zkSync community by following @zkSyncDevs on Twitter and joining the #dev-general channel in the Discord.

**Comment by zm-moon** [comment]
> Greetings devs! Welcome to the first issue of the zkSync developer update. In this monthly series we will showcase different updates about developer tools, integrations, tutorials, developer-oriented content and more.
> 
> Here are the updates for the last month:
> 
> **🛠️ Developer tools**
> 
> * Etherscan launch: the most popular block explorer is [now available on zkSync](https://twitter.com/zksync/status/1744343399417737266). Start exploring [zkSync on Etherscan](https://era.zksync.network/)
> * Remix plugin: in collaboration with Nethermind, we’ve released a plugin for the beloved browser-based IDE so you can build and deploy contracts from your browser. [Read the guide to use the plugin](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3)
> * Viem zkSync support: `viem/zksync` contains methods to send transactions or interact with a contract via paymasters and have gas fees covered! Stay tuned for more zkSync support for viem. [Check out the viem zkSync docs](https://viem.sh/zksync)
> * zkSync Sepolia testnet integrations: more developer tools now support the zkSync Sepolia testnet:
>   
>   * [theGraph CLI](https://www.npmjs.com/package/@graphprotocol/graph-cli): create and deploy subgraphs
>   * [BlockPI](https://public.blockpi.io/), [Chainstack](https://chainstack.com/build-better-with-zksync-era/) and [QuickNode](https://www.quicknode.com/chains/zksync) offer RPC nodes.
>   * Both [Atlas](https://zksync.atlaszk.com/ide) and [Remix](https://remix.ethereum.org/) allow you to build and deploy smart contracts from your browser
> 
> **📖 Documentation**
> 
> * New [hello world tutorial](https://docs.zksync.io/build/quick-start/hello-world.html): using `zksync-cli` to deploy a contract on zkSync in less than 5 minutes.
> * New [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) with everything you need to know about `zksync-ethers` and how to integrate paymasters in your dApps.
> * We also released multiple updates in the zkSync developer documentation, including design improvements and a content restructure. [Check it out here](https://docs.zksync.io/) and let us know what you think!
> 
> **📺 Content**
> 
> * [In memory node video tutorial](https://www.youtube.com/watch?v=hcavBfyh8JE): learn how to use the in-memory-node to test your contracts and debug transactions.
> * [Atlas IDE walkthrough tutorial](https://www.youtube.com/watch?v=PjMXPz2HIoY) with Samee
> * New shorts uploaded to the [zkSync Youtube channel](https://www.youtube.com/@zkSync-era/shorts?sub_confirmation=1)
> 
> **🔗 Stay connected**
> 
> * [Follow @zkSyncDevs on X](https://twitter.com/zkSyncDevs)
> * Join the `#dev-general` channel in our Discord
> 
> Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.
> 
> Keep buidling! 👩‍💻🧑🏻‍💻

Very nice update! Looking forward to more features in the future
